osu.Game/Skinning/LegacySkinExtensions.cs

```diff
@@ -88,18 +92,31 @@ IEnumerable<Texture> getTextures(ISkin skin)
             {
                 for (int i = 0; true; i++)
                 {
-                    Texture? texture;
+                    var texture = skin.GetTexture(getFrameName(i), wrapModeS, wrapModeT);
 
-                    if ((texture = skin.GetTexture(getFrameName(i), wrapModeS, wrapModeT)) == null)
+                    if (texture == null)
                         break;
 
+                    if (maxSize != null)
+                        texture = texture.WithMaximumSize(maxSize.Value);
+
                     yield return texture;
                 }
             }
 
             string getFrameName(int frameIndex) => $"{componentName}{animationSeparator}{frameIndex}";
         }
 
+        public static Texture WithMaximumSize(this Texture texture, Vector2 maxSize)
+        {
+            if (texture.DisplayWidth <= maxSize.X && texture.DisplayHeight <= maxSize.Y)
+                return texture;
+
+            // use scale adjust property for downscaling the texture in order to meet the specified maximum dimensions.
+            texture.ScaleAdjust *= Math.Max(texture.DisplayWidth / maxSize.X, texture.DisplayHeight / maxSize.Y);
```

Comment:
We're going to have to be very conscious about this, to ensure we don't have any local usages overriding `ScaleAdjust` themselves.

I've checked all usages in the project right now and it seems okay. For now.

Reply:
On a second thought, an alternative way to achieve this could be to have something like a "maximum texture size" property in `SkinnableSprite`, and if the sprite exceeds that property then the scale of it is adjusted accordingly.

That might be better as to avoid tinkering around with low properties like `Texture.ScaleAdjust`. Also saves from having to add extra properties to `GetTexture`/`GetAnimation`. @peppy thoughts? would it be worth the effort?